### PR TITLE
[1LP][RFR] Added test_service_link for MyService

### PR DESCRIPTION
**Pull request by Yadnyawalkya**
Automation test for [BZ1443772](https://bugzilla.redhat.com/show_bug.cgi?id=1443772): Clicking service link of VMDetails page from test_myservice 

{{pytest: cfme/tests/services/test_myservice.py -v -k 'test_service_link' --long-running }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 97e52bd5a911d770f467e00804a8ead265a5ef5c:

`cfme/tests/services/test_myservice.py`:
- :bangbang: Line 137:61: **E262** *inline comment should start with '# '*
- :bangbang: Line 140:40: **E261** *at least two spaces before inline comment*
- :bangbang: Line 140:41: **E262** *inline comment should start with '# '*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/services/myservice/ui.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from navmazing import NavigateToAttribute, NavigateToSibling
 from widgetastic.widget import Text, View
-from widgetastic_manageiq import Accordion, ManageIQTree, Calendar, SummaryTable
+from widgetastic_manageiq import Accordion, ManageIQTree, Calendar, SummaryTable, \
```
Comment (quarckster) [inline cfme/services/myservice/ui.py]: **REQUIRED**
Please follow style guides http://cfme-tests.readthedocs.io/guides/dev_guide.html#code-style. 

> Use parentheses () for line continuation:

File: cfme/services/myservice/ui.py [hunk 2]
```diff
@@ -10,6 +11,7 @@
 from cfme.utils.appliance import current_appliance
 from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to, ViaUI
 from cfme.utils.wait import wait_for
+from cfme.common.vm_views import VMDetailsEntities
```
Comment (quarckster) [inline cfme/services/myservice/ui.py]: **REQURIED**

> In addition to being broken up into the three sections of standard library, third-party, and the local application, imports should be sorted alphabetically. ‘import’ lines within those sections still come before ‘from ... import’ lines

File: cfme/services/myservice/ui.py [hunk 3]
```diff
@@ -147,6 +150,17 @@ def is_displayed(self):
         )
 
 
+class ServiceVMDetailsView(VMDetailsEntities):
+    """is_displayed check for cfme.tests.services.test_myservice.test_service_link"""
```
Comment (quarckster) [inline cfme/services/myservice/ui.py]: This description is not needed.

File: cfme/services/myservice/ui.py [hunk 4]
```diff
@@ -339,3 +353,13 @@ class MyServiceReconfigure(CFMENavigateStep):
 
     def step(self):
         self.prerequisite_view.configuration.item_select('Reconfigure this Service')
+
+
+@navigator.register(MyService, 'VMDetails')
+class MyServiceVMDetails(CFMENavigateStep):
+    VIEW = ServiceVMDetailsView
+
```
Comment (quarckster) [inline cfme/services/myservice/ui.py]: **REQUIRED**
Please remove this blank line

File: cfme/tests/services/test_myservice.py [hunk 5]
```diff
@@ -123,3 +124,18 @@ def test_download_file(appliance, context, needs_firefox, myservice, filetype):
     with appliance.context.use(context):
         myservice = MyService(appliance, name=service_name, vm_name=vm_name)
         myservice.download_file(filetype)
+
+
+# Automation test for BZ1443772
+@pytest.mark.parametrize('context', [ViaUI])
+def test_service_link(appliance, context, myservice):
+    """Tests service link from VM details page"""
+
```
Comment (quarckster) [inline cfme/tests/services/test_myservice.py]: Please remove this blank line

File: cfme/tests/services/test_myservice.py [hunk 6]
```diff
@@ -123,3 +124,18 @@ def test_download_file(appliance, context, needs_firefox, myservice, filetype):
...
+    service_name, vm_name = myservice
+    with appliance.context.use(context):
+        myservice = MyService(appliance, name=service_name, vm_name=vm_name)
+        earlier_title = 'Service "' + service_name + '"'   # Page title before navigation
```
Comment (quarckster) [inline cfme/tests/services/test_myservice.py]: **REQUIRED**
Please use `format`.

File: cfme/tests/services/test_myservice.py [hunk 7]
```diff
@@ -123,3 +124,18 @@ def test_download_file(appliance, context, needs_firefox, myservice, filetype):
     with appliance.context.use(context):
         myservice = MyService(appliance, name=service_name, vm_name=vm_name)
         myservice.download_file(filetype)
+
+
+# Automation test for BZ1443772
```
Comment (quarckster) [inline cfme/tests/services/test_myservice.py]: **REQURIED**
This comment must be placed within test's docstring.

File: cfme/tests/services/test_myservice.py [hunk 8]
```diff
@@ -123,3 +124,18 @@ def test_download_file(appliance, context, needs_firefox, myservice, filetype):
...
+@pytest.mark.parametrize('context', [ViaUI])
+def test_service_link(appliance, context, myservice):
+    """Tests service link from VM details page"""
+
+    service_name, vm_name = myservice
+    with appliance.context.use(context):
+        myservice = MyService(appliance, name=service_name, vm_name=vm_name)
+        earlier_title = 'Service "' + service_name + '"'   # Page title before navigation
+        view = navigate_to(myservice, 'VMDetails')
+        view.relationships.click_at("Service")
```
Comment (quarckster) [inline cfme/tests/services/test_myservice.py]: **REQUIRED**
I guess here the has been changed. So you need to create a new respectful view and instead asserting titles you can just assert `is_displayed` property.

**Comment by mshriver** [comment]
@Yadnyawalkya Check out the TODO in the check_vm_add method, where web_ui.quadicon is used, you can replace that with the quadicon widget you've defined in this PR.  I've got a few updates to this class as well in #5401 from some things I found auditing tests and templates.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 823b9883b28f10a534d94c96082fb2d12abe27b0:

`cfme/services/myservice/ui.py`:
- :red_circle: Line 3:1: **F401** *'widgetastic_manageiq.BaseNonInteractiveEntitiesView' imported but unused*
- :red_circle: Line 164:28: **F821** *undefined name 'VMDetailsEntities'*

Please, rectify these issues :smirk: .

* 789ffb4be84f5bb8d6e089d298fc74bca14dbb60
  * Merge commit detected!!! :bangbang:
* 823b9883b28f10a534d94c96082fb2d12abe27b0
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by mshriver** [comment]
Rebase, no merge commits please

**Comment by dajoRH** [comment]
Lint report for commit 070276c0f03be592e59cbc26887d7712a5d16413:

`cfme/services/myservice/ui.py`:
- :red_circle: Line 3:1: **F401** *'widgetastic_manageiq.BaseNonInteractiveEntitiesView' imported but unused*
- :red_circle: Line 164:28: **F821** *undefined name 'VMDetailsEntities'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by Yadnyawalkya** [comment]
No problem, surely do @izapolsk! 
Thanks @izapolsk  and @quarckster.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by izapolsk** [comment]
@Yadnyawalkya, sure, I'm ready to merge it but it has to be rebased.

**Pull request merged**